Polybar module

cdutsov opened the pull request:
Hello,

thanks for the suggestion to include the script! This is my first PR actually, so I hope I'm doing it right.

I have changed the script to have options for the colors and the polybar ipc hook name. The colors abvailable for customization are for the current open workspace and all workspaces on external monitors. 
I have also added the polybar updater and polybar module scripts in the scripts folder and in the setup.py [1].

For the module to work, the necessary things are to add this in the i3/config:
```exec --no-startup-id i3-autoname-workspaces
exec --no-startup-id i3-groups-polybar-module-updater
```

And this in the polybar config:
```
[module/i3-mod]
type = custom/ipc
hook-0 = i3-groups-polybar-module -c "#a3be8c" -e "#EBCB8B" -k "i3-mod 1"
initial = 1
```

I have tested this and it seems to be working. If you need me to change or explain something, please don't hesitate to ask!
 

[1]: Ok, I think I added them, but I don't see them in the pull request. Maybe I did something wrong with the branching. Sorry, I am not very used to git still.

infokiller commented:
Thanks for the PR! `setup.py` is included in it, don't worry.
If you can fix the [code factor warnings](https://www.codefactor.io/repository/github/infokiller/i3-workspace-groups/pull/40) that would be great. You need to do these changes in new commits, then push them to your fork, and then they'll appear in the PR automatically.


I'll also do a thorough review and local testing later (before I publish it with the next version), along with some docs, but I don't want to bother you with this for now (and I also don't have much time now and didn't want to block this, it's a good contribution).

cdutsov commented:
Hi! I fixed the errors in the CodeFactor. I have also added an option `-s` to switch between the shorthand and normal notaiton for the workspaces. The shorthand is this, for example:
![image](https://user-images.githubusercontent.com/11442558/99898058-e1d1a580-2c9e-11eb-9f62-9f8c2dad41ba.png)
and the normal is this:
![image](https://user-images.githubusercontent.com/11442558/99898063-eeee9480-2c9e-11eb-81f3-ec332a26fca0.png)

If you need me to write some explanations or more comments in the code, please tell me.

infokiller commented:
Thanks a lot!
I'll merge this now, but as I said before, I'll probably do some changes later when I have more time to work on the project (and maybe I'll just migrate all the code and scripts to golang...).
